Cargo.toml

```diff
@@ -16,14 +16,15 @@ travis-ci = { repository = "KokaKiwi/rust-hex", branch = "master" }
 
 [features]
 default = ["std"]
-std = []
+alloc = ["serde/alloc"]
+std = ["alloc", "serde/std"]
```

Comment:
I tried that on my local dev env and i ran into a feature-resolutions problem: Enabling the std feature implicitly enables the serde feature because of the dependency on serde/std feature...
I think it's kinda a Cargo-level issue and i don't really know what to do about that.

Letting it be would make hex compiling implicitly serde by default even if it's should be an optional feature, even if not actually used, so shouldn't be a problem by itself apart from the fact it actually compiles something unused...

Reply:
the relevant issue is https://github.com/rust-lang/cargo/issues/3494

Reply:
I would tend toward merging this PR, because it fixes some outstanding issues.

The bug does not affect the run time performance and only increases the compile-time, which is kinda neglect able.

Though for now I think waiting is the best option, especially because this would bump the dependencies from zero to one. (some people specifically use this crate, because it has zero dependencies)

Reply:
Unfortunately, waiting also means that users of this crate in environments without alloc depend on a git branch rather than a released version until then. (No problem for the application I'm writing currently, though, as it's just a demo that sits in git and is not aim for to crates.io.)

Would introducing a feature "serde-std" that is to be enabled by users who want to use the crate with both serde and std resolve the issue? (Or a "serde-nostd" that only depends on serde and not on serde/std).

Reply:
I was thinking, what about simply removing the dependencies over `serde/std` and `serde/alloc`, just having dependencies over "nostd-serde"?
`hex` does not really *need* the std version of `serde` as it only requires the de/serializing traits
Also, it would let the lib user being the only one setting the deps as they require it or not